src/frontend/src/handler/alter_streaming_rate_limit.rs

```diff
@@ -97,14 +101,46 @@ pub async fn handle_alter_streaming_rate_limit(
             (StatementType::ALTER_TABLE, table.id.table_id)
         }
         PbThrottleTarget::Sink => {
-            let reader = session.env().catalog_reader().read_guard();
-            let (table, schema_name) =
-                reader.get_sink_by_name(db_name, schema_path, &real_table_name)?;
-            if table.target_table.is_some() {
-                bail!("ALTER SINK_RATE_LIMIT is not for sink into table")
+            let sink_id = {
+                let reader = session.env().catalog_reader().read_guard();
+                let (table, schema_name) =
+                    reader.get_sink_by_name(db_name, schema_path, &real_table_name)?;
+
+                if table.target_table.is_some() {
+                    bail!("ALTER SINK_RATE_LIMIT is not for sink into table")
+                }
+                session.check_privilege_for_drop_alter(schema_name, &**table)?;
+                table.id.sink_id
+            };
+            let selection = Some(Expr::BinaryOp {
+                left: Box::new(Expr::Identifier(SINK_ID.into())),
+                op: BinaryOperator::Eq,
+                right: Box::new(Expr::Value(Value::Number(sink_id.to_string()))),
+            });
+            let query = gen_query_from_table_name_with_selection(
+                ObjectName::from(vec![
+                    Ident::from("rw_catalog"),
+                    Ident::from("rw_sink_decouple"),
```

Comment:
Checking the system table here seems a bit strange. I think it is better to do the check in meta [update_sink_rate_limit_by_job_id](https://github.com/risingwavelabs/risingwave/blob/5a972a6704769bc819744b0a91c92613dba9ef5a/src/meta/src/controller/streaming_job.rs#L1666). You can check the `SinkLogStoreType` in the sink plan node there.